core/src/main/java/org/geysermc/geyser/entity/type/TextDisplayEntity.java

```diff
@@ -60,6 +67,15 @@ protected void initializeMetadata() {
     }
 
     public void setText(EntityMetadata<Component, ?> entityMetadata) {
+        this.text = entityMetadata.getValue();
         this.dirtyMetadata.put(EntityDataTypes.NAME, MessageTranslator.convertMessage(entityMetadata.getValue()));
+        calculateLines();
+    }
+
+    private void calculateLines() {
+        if (text == null) {
+            return;
```

Comment:
Given that this is called after the text is set, the line count should probably be reset here.